viewing_party/party.py

```diff
@@ -0,0 +1,235 @@
+        watched_list = friends[friend]["watched"]
+        for movies in watched_list:
+            for individual_title in movies:
+                #print(movies[individual_title])
+                friends_movies.add(movies[individual_title])
+                
+    #print(friends_movies)
+
+    users_movies = set(users_movies)
+    
+    movies_only_user_watched = users_movies - friends_movies
+    #print(movies_only_user_watched)
+    
+    final_movie_list_of_dictionaries = []
+    for movie in movies_only_user_watched:
+        final_movie_list_of_dictionaries.append({"title": movie}) 
+    
+    return final_movie_list_of_dictionaries
+
+def get_friends_unique_watched(user_data):
+    #first, let's get the list of movies the user has watched
+    users_movies = []
+    for index in range(len(user_data["watched"])):    
+        users_movies.append(user_data["watched"][index]["title"])
+
+    #next, let's get the set of movies the friends have watched
+    friends_movies = set()
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movies in watched_list:
+            for individual_title in movies:
+                #print(movies[individual_title])
+                friends_movies.add(movies[individual_title])
+                
+    #convert the users_movies to a set for comparison
+    users_movies = set(users_movies)
+    
+    #find the difference
+    movies_at_least_one_friend_watched = friends_movies - users_movies
+
+    #convert to a list of dictionaries to return
+    final_movie_list_of_dictionaries = []
+    for movie in movies_at_least_one_friend_watched:
+        final_movie_list_of_dictionaries.append({"title": movie}) 
+    
+    return final_movie_list_of_dictionaries
+
+def get_available_recs(user_data):
+
+    #Determine a list of recommended movies. 
+    # A movie should be added to this list if and only if
+    #The user has not watched it
+    #At least one of the user's friends has watched
+    #The "host" of the movie is a service that is in the user's "subscriptions"
+
+    user_watched_list = user_data["watched"]
+    user_subscriptions = user_data["subscriptions"]
+    
+    recommended_movies = []
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movie in watched_list:
+            if movie["host"] in user_subscriptions and movie not in recommended_movies:
+                if len(user_data['watched']) > 0:
+                    count = 0
+                    for title in user_data["watched"]:
+                        if movie['title'] is not title['title']:
+                            recommended_movies.append(movie)
+                        else:
+                            count += 1
+
+                        if count == len(recommended_movies):
+                            recommended_movies = []      
```

Comment:
There are a few logical bugs in this section.  1: starting on line 181, if there are multiple titles in the user_data["watched"] list that are not equal to the current movie, the movie will be added to the list multiple times.  2: recommended_movies will be overwritten by the test on line 187 in very specific cases (ie, if there are 2 movies in user_data["watched"], and recommended_movies starts empty, and a movie is added that matches one of those.  This makes one of the tests pass, but will cause problems in other cases.)